src/cirq_iqm/devices/iqm_device_metadata.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2020–2022 Cirq on IQM developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""DeviceMetadata subtype for IQM devices."""
+from __future__ import annotations
+
+from typing import FrozenSet, Optional, Type, Union
+
+import cirq
+from cirq import NamedQubit, devices, ops
+from cirq.contrib.routing.router import nx
+from iqm_client import QuantumArchitectureSpecification
+
+# Mapping from IQM operation names to cirq operations
+_IQM_CIRQ_OP_MAP: dict[str, Optional[tuple[Union[Type[cirq.Gate], cirq.Gate, cirq.GateFamily], ...]]] = {
+    # XPow and YPow kept for convenience, Cirq does not know how to decompose them into PhasedX
+    # so we would have to add those rules...
+    'phased_rx': (cirq.ops.PhasedXPowGate, cirq.ops.XPowGate, cirq.ops.YPowGate),
+    'cz': (cirq.ops.CZ,),
+    'measurement': (cirq.ops.MeasurementGate,),
+    'barrier': None,
+}
+
+
+@cirq.value.value_equality
+class IQMDeviceMetadata(devices.DeviceMetadata):
+    """Hardware metadata for IQM devices.
+
+    Args:
+        qubits: frozenset of qubits that exist on the device
```

Comment:
```suggestion
        qubits: qubits that exist on the device
```
Types should not be repeated in the descriptions if possible, since they will be clear from the type annotations.